platform-sdk/platform-apps/tests/ConsistencyTestingTool/src/main/java/com/swirlds/demo/consistency/ConsistencyTestingToolMain.java

```diff
@@ -45,6 +53,31 @@ public class ConsistencyTestingToolMain implements SwirldMain {
      */
     private static final SoftwareVersion softwareVersion = new BasicSoftwareVersion(1);
 
+    static {
+        try {
+            logger.info(STARTUP.getMarker(), "Registering ConsistencyTestingToolState with ConstructableRegistry");
+            ConstructableRegistry constructableRegistry = ConstructableRegistry.getInstance();
+            constructableRegistry.registerConstructable(new ClassConstructorPair(
+                    ConsistencyTestingToolState.class,
+                    () -> new ConsistencyTestingToolState(
+                            NO_OP_MERKLE_STATE_LIFECYCLES, version -> new BasicSoftwareVersion(version.major()))));
+            logger.info(STARTUP.getMarker(), "ConsistencyTestingToolState is registered with ConstructableRegistry");
+            registerPlatformClasses(constructableRegistry);
+
+        } catch (ConstructableRegistryException e) {
+            logger.error(STARTUP.getMarker(), "Failed to register ConsistencyTestingToolState", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void registerPlatformClasses(ConstructableRegistry constructableRegistry) {
+        logger.info(STARTUP.getMarker(), "Registering PlatformState classes with ConstructableRegistry");
+        final var schema = new V0540PlatformStateSchema();
+        final StateDefinition def = schema.statesToCreate().iterator().next();
+        final var md = new StateMetadata<>(PlatformStateService.NAME, schema, def);
+        registerWithSystem(md, constructableRegistry);
+    }
```

Comment:
Is there a reason not to use the `lifecycles` implementation for this registration? In the AB->Roster refactoring fixes, I have to initialize the RosterState/RosterMap in addition to the PlatformState in each of such states. Having this logic duplicated in multiple places makes this hard, and also can lead to discrepancies/bugs. It would be best to consolidate this logic in a single place (such as the `lifecycles` class or similar).

This comment applies to similar code in other testing state classes in this fix.

Reply:
The reason was that `NO_OP_MERKLE_STATE_LIFECYCLE` didn't take care of all necessary class registrations that I extracted to `StateUtils#registerWithSystem`. However, you have a good point - they should not be separated. So, I add a call of `StateUtils#registerWithSystem` into `NoOpMerkleStateLifecycles#initPlatformState` and removed the logic duplication